### ISSUE TITLE
Search form — fix inappropriate use of label elements for date inputs plus the group of checkboxes for record type

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/FreeBMD2#490]
HTML label elements and form inputs (fields) should be linked together in pairs. For those who can see the form, failure to do this is not an issue. However, for those who rely on screen reader software to interpret the form for them, such failures are confusing at best.

We are using HTML labels to describe groups of form inputs which does not work for a user who cannot see the form. The HTML element that groups inputs is the `fieldset`: the element that describes (or labels) a `fieldset` is the `legend`. We have styles for both these elements to achieve the desired visual effect.

(It is not ideal to use nested fieldsets, but it is better than misusing labels.)

**Comment by AlOneill** [comment]
**Looking at the group of checkboxes headed by 'Record Type':**

The main problem is the label element, Record Type, that does not have an associated input. It is the wrong element for the task of identifying a *group* of checkboxes.

I suggest that we follow the markup pattern used on REG:

* use a `fieldset.choices` immediately inside the `li` that holds "Record type" plus the checkboxes — the class will remove the border and adjust the spacing

* replace the `label` element, Record Type, with a `legend.label`, Record Type

* after the `legend`, open a `ol.choices-group` which will remove the numbers from the list items— it will **not** need to be a grid

* the `li`s for the checkboxes will **not** need to be grid__items

The class `.choices` and the related classes are in `freereg_content.css`. 
None of the inputs need `tabindex="0"`  and instances of `.ttip` can be removed

A second issue is that there are some 29 instances of tabindex="0" in the form — only labels with tooltips need these. An input is naturally tab-able.

**Comment by AlOneill** [comment]
I have been asked to clarify the issue described above. I will try.

This image shows the group of record type inputs as they are currently marked up:

![record-group-issues.png](https://images.zenhubusercontent.com/5ce6b34165b1a70acc0a9124/3470ab4d-3b0f-433f-b75e-e195ad062bb4)

"Record type" is marked up as an HTML label. This is incorrect because it does not go with a single form control (input). The four record type inputs — Select All, Birth, Death, Marriage — each have their own HTML label which _is_ correct.

The four record type inputs are a group of inputs. The HTML element for grouping form inputs is the `fieldset`. It has its own labelling element, the `legend`. We can preserve the current appearance of such a fieldset by giving it the class `choices` which removes the border and adjusts the spacing. Then give the legend the _class_ of label. With those changes, the group of inputs will look like this (the spacing after the legend can be adjusted if necessary):

![record-group-fieldset.png](https://images.zenhubusercontent.com/5ce6b34165b1a70acc0a9124/e7ebbbf3-7ba9-4264-9534-df63eb1b7de4)

There is now no need for a complicated grid with grid-items to control the placement of the four inputs. Simply open a nested, ordered list with the class `choices-group`. All the classes needed are in `freereg_content.css` which is already available on BMD.

As well as removing the grid and grid__item classes, the record type inputs have `tabindex="0"` which can be removed as unnecessary.

Any questions? Please ask away …

**A related issue**, the markup of the "From Quarter/Year" and "To Quarter/Year" groups of inputs is a similar problem with a similar solution. I can supply details if needed but am out of energy/concentration for now.

**Comment by AlOneill** [comment]
@Vino-S 
Thank you for fixing the Record types markup. Denise said that it was you.

Let me know if you need any guidance for fixing the dates markup: one label for each input is the rule.

**Comment by KristinaGadzhieva** [comment]
@DeniseColbert @AlOneill Could you please check it on test 3?

**Comment by AlOneill** [comment]
The markup for the various date inputs now looks fine to me — thank you! 

If I feel better later I will check that the announcements make sense using a screen reader — I do not anticipate any issues but it is always sensible to double check these things.

**Comment by KristinaGadzhieva** [comment]
@Vino-S ready for deployment
